manual/manual/refman/exten.etex

```diff
@@ -8,2767 +8,109 @@ that are implemented in OCaml, but not described in chapter \ref{c:refman}.
-  end in
-  let pair = (module P : Typ.PAIR with type t = s1 * s2) in
-  Typ.Pair pair
-\end{caml_example*}
-
-And finally, here is an example of a polymorphic function that takes the
-runtime representation of some type "'a" and a value of the same type,
-then pretty-prints the value into a string:
-
-\begin{caml_example*}{verbatim}
-open Typ
-let rec to_string: 'a. 'a Typ.typ -> 'a -> string =
-  fun (type s) t x ->
-    match t with
-    | Int eq -> Int.to_string (TypEq.apply eq x)
-    | String eq -> Printf.sprintf "%S" (TypEq.apply eq x)
-    | Pair p ->
-        let module P = (val p : PAIR with type t = s) in
-        let (x1, x2) = TypEq.apply P.eq x in
-        Printf.sprintf "(%s,%s)" (to_string P.t1 x1) (to_string P.t2 x2)
-\end{caml_example*}
-
-Note that this function uses an explicit polymorphic annotation to obtain
-polymorphic recursion.
-\fi
+\input{firstclassmodules.tex}
 
 \section{s:module-type-of}{Recovering the type of a module}
 %HEVEA\cutname{moduletypeof.html}
-
 \ikwd{module\@\texttt{module}}
 \ikwd{type\@\texttt{type}}
 \ikwd{of\@\texttt{of}}
 \ikwd{include\@\texttt{include}}
-
-(Introduced in OCaml 3.12)
-
-\begin{syntax}
-module-type:
-     ...
-   | 'module' 'type' 'of' module-expr
-\end{syntax}
-
-The construction @'module' 'type' 'of' module-expr@ expands to the module type
-(signature or functor type) inferred for the module expression @module-expr@.
-To make this module type reusable in many situations, it is
-intentionally not strengthened: abstract types and datatypes are not
-explicitly related with the types of the original module.
-For the same reason, module aliases in the inferred type are expanded.
-
-A typical use, in conjunction with the signature-level @'include'@
-construct, is to extend the signature of an existing structure.
-In that case, one wants to keep the types equal to types in the
-original module. This can done using the following idiom.
-\begin{caml_example*}{verbatim}
-module type MYHASH = sig
-  include module type of struct include Hashtbl end
-  val replace: ('a, 'b) t -> 'a -> 'b -> unit
-end
-\end{caml_example*}
-The signature "MYHASH" then contains all the fields of the signature
-of the module "Hashtbl" (with strengthened type definitions), plus the
-new field "replace".  An implementation of this signature can be
-obtained easily by using the @'include'@ construct again, but this
-time at the structure level:
-\begin{caml_example*}{verbatim}
-module MyHash : MYHASH = struct
-  include Hashtbl
-  let replace t k v = remove t k; add t k v
-end
-\end{caml_example*}
-
-Another application where the absence of strengthening comes handy, is
-to provide an alternative implementation for an existing module.
-\begin{caml_example*}{verbatim}
-module MySet : module type of Set = struct
-  include Set[@@ellipsis]
-end
-\end{caml_example*}
-This idiom guarantees that "Myset" is compatible with Set, but allows
-it to represent sets internally in a different way.
+\input{recoveringtype.tex}
```

Comment:
Is there a reason to change the name compared to the cutname?

Reply:
I will fix these in a moment.

Reply:
Why do some sections (e.g recursive-modules) not have a cutname anyway?

Reply:
They all should have a cutname and indeed the one for recursive modules is missing, leading to an anonymous page name: https://ocaml.org/releases/4.11/htmlman/manual024.html . 
This PR is probably the right time to fix this issue.